module/main.go

```diff
@@ -21,7 +22,14 @@ var (
 )
 
 func main() {
-	utils.ContextualMain(mainWithArgs, golog.NewLogger("rplidarModule"))
+	utils.ContextualMain(mainWithArgs, NewLoggerFromArgs("rplidarModule"))
+}
+
+func NewLoggerFromArgs(moduleName string) golog.Logger {
+	if len(os.Args) >= 3 && os.Args[2] == "--log-level=debug" {
```

Comment:
1. I recommend we not care where in the os.Args list the --log-level parameter is
Something more like this:
```
import (
	"os"
	"strings"
)

func hasLogLevel(arg string) bool {
	return strings.HasPrefix(arg, "-log-level=") || strings.HasPrefix(arg, "--log-level=")
}

func isDebug(args []string) bool {
	for _, arg := range args {
		if hasLogLevel(arg) {
        parsed := strings.Split(arg, "-log-level=")
        return parsed[1] == "debug"
		}
	}
  return false
}

if (isDebug) {
return golog.NewDebugLogger(moduleName)
}
return golog.NewDevelopmentLogger(moduleName)
```

2. I see that the [glog](https://github.com/edaniels/golog/blob/master/logger.go) package doesn't have convenience functions for setting log level. Do we know why that is? It seems strange to me that the module framework allows the user to specify a log level but we don't have support in our logging library to manually specify that we only want to log error, warn etc. Could we check with eric & the netcode team regarding how module teams are expected to handle this 'log-level' param?

Reply:
I have switched to using the builtin function similar to how it is done in this PR: https://github.com/viamrobotics/rdk/pull/2373/files#diff-e7ba4c8e22457875e841fc23dd1efa585bed7f88fcc37e260886e80a7edb9ab7

Let me know if that answers question 2.

As for question 1 I think we'd want to discuss with the netcode team regarding if we want to improve the builtin function or create a custom one ourselves